app/src/main/java/org/systers/mentorship/view/fragments/RequestPagerFragment.kt

```diff
@@ -45,7 +45,31 @@ class RequestPagerFragment: BaseFragment() {
             requestsList = it.getParcelableArrayList(Constants.REQUEST_LIST)
             emptyListText = it.getString(Constants.REQUEST_EMPTY_LIST_TEXT)
         }
+        setUI()
+    }
+
+    private val openRequestDetail: (Relationship) -> Unit =
+            { requestDetail ->
+                val intent = Intent(activity, RequestDetailActivity::class.java)
+                intent.putExtra(Constants.RELATIONSHIP_EXTRA, requestDetail)
+                startActivityForResult(intent, 1000)
+            }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        val id = data!!.getStringExtra("requestid")
+        val iter = requestsList.iterator()
+
+        while (iter.hasNext()) {
+            val reqId = iter.next().id
+            if(reqId.toString() == id) {
+                iter.remove()
+            }
+        }
+        setUI()
+    }
 
+    fun setUI() {
```

Comment:
```suggestion
    private fun setView() {
```

cc @m-murad

Reply:
I think as it is also used in both classes, we should put it in Constants class .
cc @isabelcosta @mohak1283